### PR TITLE
Fix broken links on std::boxed doc page

### PR DESCRIPTION
**Pull request by ollie27**
r? @QuietMisdreavus

### CODE REVIEW

**Review by QuietMisdreavus — APPROVED** [review]

File: src/liballoc/lib.rs [hunk 1]
```diff
@@ -65,7 +65,7 @@
 #![allow(explicit_outlives_requirements)]
 
 #![warn(deprecated_in_future)]
-#![warn(intra_doc_link_resolution_failure)]
+#![deny(intra_doc_link_resolution_failure)]
```
Comment (RalfJung) [inline src/liballoc/lib.rs]: We compile with `-D warnings`, which does this even make a difference?

Comment (ollie27) [reply src/liballoc/lib.rs]: We don't yet pass `-D warnings` to rustdoc when building the docs unfortunately.

**Comment by ollie27** [comment]
Looks like I was premature with changing `intra_doc_link_resolution_failure` lint back to deny as we now have some failures from `stdsimd` https://github.com/rust-lang/rust/pull/59868#issuecomment-481976037 so I've removed the lint changes.

File: src/liballoc/lib.rs (hunk 1 above)
Comment (RalfJung) [reply src/liballoc/lib.rs]: Ah I see. When/if you change this next time, please add a comment explaining that. :)

**Pull request merged**